# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserBridgeChild.cpp
# Commit: 2d0e5336e455
# Full Hash: 2d0e5336e455f560cdd345afa57f21b8cdbfc1fb
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2019-08-26 22:30:36
# Regressor Bug: 1563619
# File Overlap Count: 1
# Description:
#   Bug 1563619 - Handle subframe crashes in BrowserParent::ActorDestroy. r=mconley,kmag,peterv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40937
# ==============================================================================

diff -r fd3185644de9 -r 2d0e5336e455 dom/ipc/BrowserBridgeChild.cpp
--- a/dom/ipc/BrowserBridgeChild.cpp	Mon Aug 26 13:07:26 2019 +0000
+++ b/dom/ipc/BrowserBridgeChild.cpp	Mon Aug 26 13:08:32 2019 +0000
@@ -10,6 +10,7 @@
 #endif
 #include "mozilla/dom/BrowserBridgeChild.h"
 #include "mozilla/dom/BrowsingContext.h"
+#include "mozilla/dom/MozFrameLoaderOwnerBinding.h"
 #include "nsFocusManager.h"
 #include "nsFrameLoader.h"
 #include "nsFrameLoaderOwner.h"
@@ -195,6 +196,24 @@
   return IPC_OK();
 }
 
+mozilla::ipc::IPCResult BrowserBridgeChild::RecvSubFrameCrashed(
+    BrowsingContext* aContext) {
+  if (aContext) {
+    RefPtr<nsFrameLoaderOwner> frameLoaderOwner =
+        do_QueryObject(aContext->GetEmbedderElement());
+    IgnoredErrorResult rv;
+    RemotenessOptions options;
+    options.mError.Construct(static_cast<uint32_t>(NS_ERROR_FRAME_CRASHED));
+    frameLoaderOwner->ChangeRemoteness(options, rv);
+
+    if (NS_WARN_IF(rv.Failed())) {
+      return IPC_FAIL(this, "Remoteness change failed");
+    }
+  }
+
+  return IPC_OK();
+}
+
 void BrowserBridgeChild::ActorDestroy(ActorDestroyReason aWhy) {
   mIPCOpen = false;
 